### PR TITLE
Add virtual members test

### PR DESCRIPTION
**Pull request by Szpoti**
Probably brought the latest master changes as well in the first PR, this should be better.

### CODE REVIEW

File: WalletWasabi.Tests/UnitTests/Transactions/SmartTransactionTests.cs [hunk 1]
```diff
@@ -203,6 +204,30 @@ public void SmartTransactionLineDeserialization()
 		}
 	}
 
+	[Fact]
+	public void SmartTransactionVirtualMembersEqualsTest()
```
Comment (nopara73) [inline WalletWasabi.Tests/UnitTests/Transactions/SmartTransactionTests.cs]:
```suggestion
	public void SmartTransactionVirtualMembersEquals()
```

**Review by nopara73 — APPROVED** [review]
Very good, but we need moaaaaar tests

**Comment by nopara73** [comment]
@Szpoti can you add 6 more tests? You'll have to manipulate the transaction object by hand to create various scenarios. Don't be afraid!

**Comment by Szpoti** [comment]
Sure, but can you elaborate on your vision what/where to manipulate for testing?

**Comment by nopara73** [comment]
Sure. Play with

- input/output with the same scriptPubKey
- input/output with different scriptPubKey, but on the same pubKey

Then test when WalletVirtualOutputs and ForeignVirtualOutputs share (a) scriptPubKey, (b) pubKey

**Pull request merged**